File: app/router/process.go
package router

import (
	"github.com/aihop/gopanel/app/api"
	"github.com/aihop/gopanel/app/middleware"
	"github.com/aihop/gopanel/constant"
	"github.com/aihop/gopanel/pkg/websocket"
	"github.com/gofiber/fiber/v3"
)

func ProcessRouter(r fiber.Router) {
	processRouter := r.Group("process")
	processRouter.Use(middleware.JWT(constant.UserRoleAdmin))
	{
		// 系统进程监控
		processRouter.Get("/ws", websocket.New(api.ProcessWs))
		processRouter.Post("/list", api.ListProcess)
		processRouter.Post("/stop", api.StopProcess)
		processRouter.Post("/checkPort", api.CheckProcessPort)
	}
}

File: admin/src/api/modules/process.ts
import http from "@/api"
import type { Process } from "../interface/process"

export const ProcessList = (req: Process.ListReq) => {
	return http.post<any>(`/process/list`, req)
}

export const StopProcess = (req: Process.StopReq) => {
	return http.post<any>(`/process/stop`, req)
}

/** 检查端口是否被占用 */
export async function ProcessCheckPort(req: Process.PortReq) {
	return http.post<any>(`/process/checkPort`, req)
}
 
 
 

 
 

 
 

 
 
 
 
 